trunk/src/modules/par_mesh.F90

```diff
@@ -57,16 +57,28 @@ subroutine distr_mesh()
       write(6,100) 'start distr_mesh, DISTRIBUTED = ', DISTRIBUTED
    endif
    100 format(A,L2)
+!
+   if (NUM_PROCS.eq.1) return
 !
 !..1. Determine new partition
    call MPI_BARRIER (MPI_COMM_WORLD, ierr); start_time = MPI_Wtime()
    if ((ZOLTAN_LB .eq. 0) .or. (.not. DISTRIBUTED)) then
```

Comment:
I just noticed now that we have enabled the `hp3gen` routine to distribute the mesh rightaway, this line prevents the user from setting a partitioner other than `ZOLTAN_LB=0` for initial distribution.

I think we can change it simply to:

`if (ZOLTAN_LB .eq. 0) then`

Reply:
I don't think it should be changed. Some partitioners (e.g. weighted graph partitioner) require options to be set to enable starting with a non-distributed mesh. In fact many of the Zoltan partitioners return errors if there are fewer elements than processors and it's fairly common to start with single element meshes and then refine before partitioning (which you would do after `hp3gen`). Partitioning first using the simple method speeds up initial calls to `update_gdof` and `update_Ddof`, the user can repartition how they wish after that.

Reply:
I'm ok with that; in this case though, we should remove the option in problem's `set_initial_mesh` routines for the user to set the Load Balancer before `hp3gen` is called (because the choice will be simply ignored here).